mixed-content/generic/mixed-content-test-case.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * @fileoverview Test case for mixed-content in Web Platform Tests.
+ * @author burnik@google.com (Kristijan Burnik)
+ */
+
+/**
+ * MixedContentTestCase exercises all the tests for checking browser behavior
+ * when resources regarded as mixed-content are requested. A single run covers
+ * only a single scenario.
+ * @param {object} scenario A JSON describing the test arrangement and
+ *     expectation(s). Refer to /mixed-content/spec.src.json for details.
+ * @param {string} description The test scenario verbose description.
+ * @param {SanityChecker} sanityChecker Instance of an object used to check the
+ *     running scenario. Useful in debug mode. See ./sanity-checker.js.
+ *     Run {@code ./tools/generate.py -h} for info on test generating modes.
+ * @return {object} Object wrapping the start method used to run the test.
+ */
+function MixedContentTestCase(scenario, description, sanityChecker) {
+  var insecureProtocol = "http";
+  var secureProtocol = "https";
+
+  var sameOriginHost = location.hostname;
+  var crossOriginHost = "{{domains[www1]}}";
+
+  // These values can evaluate to either empty strings or a ":port" string.
+  var insecurePort = getNormalizedPort(parseInt("{{ports[http][0]}}", 10));
+  var securePort = getNormalizedPort(parseInt("{{ports[https][0]}}", 10));
+
+  var resourcePath = "/mixed-content/generic/expect.py";
+
+  // Map all endpoints to scenario for use in the test.
+  var endpoint = {
+    "same-origin":
+      location.origin + resourcePath,
+    "same-host-https":
+      secureProtocol + "://" + sameOriginHost + securePort + resourcePath,
+    "same-host-http":
+      insecureProtocol + "://" + sameOriginHost + insecurePort + resourcePath,
+    "cross-origin-https":
+      secureProtocol + "://" + crossOriginHost + securePort + resourcePath,
+    "cross-origin-http":
+      insecureProtocol + "://" + crossOriginHost + insecurePort + resourcePath
+  };
+
+  // Mapping all the resource requesting methods to the scenario.
+  var resourceMap = {
+    "a-tag": requestViaAnchor,
+    "area-tag": requestViaArea,
+    "fetch-request": requestViaFetch,
+    "form-tag": requestViaForm,
+    "iframe-tag": requestViaIframe,
+    "img-tag":  requestViaImage,
+    "script-tag": requestViaScript,
+    "worker-request": requestViaWorker,
+    "xhr-request": requestViaXhr,
+    "audio-tag": requestViaAudio,
+    "video-tag": requestViaVideo,
+    "picture-tag": requestViaPicture,
+    "object-tag": requestViaObject,
+    "link-css-tag": requestViaLinkStylesheet,
+    "link-prefetch-tag": requestViaLinkPrefetch
+  };
+
+  sanityChecker.checkScenario(scenario, resourceMap);
+
+  // Mapping all expected MIME types to the scenario.
+  var contentType = {
+    "a-tag": "text/plain",
+    "area-tag": "application/json",
+    "fetch-request": "application/json",
+    "form-tag": "text/html",
+    "iframe-tag": "text/html",
+    "img-tag":  "image/png",
+    "script-tag": "text/plain",
```

Comment:
Can you pick an executable-as-javascript MIME type? That is, one of these: https://code.google.com/p/chromium/codesearch#chromium/src/components/mime_util/mime_util.cc&rcl=1435735758&l=59